src/main/java/online/talkandtravel/security/AuthenticationFacade.java

```diff
@@ -0,0 +1,62 @@
+package online.talkandtravel.security;
+
+import lombok.extern.log4j.Log4j2;
+import online.talkandtravel.exception.user.UserNotAuthenticatedException;
+import online.talkandtravel.model.entity.User;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+/**
+ * The purpose of this class is to abstract and simplify the interaction
+ * with {@link SecurityContextHolder}, making the retrieval of the authenticated user easier and
+ * decoupling the static access to security context from the rest of the codebase
+ * To fully leverage the Spring dependency injection and be able to retrieve the authenticated user
+ * everywhere, not just in @RestController beans
+ */
+@Component
+@Log4j2
+public class AuthenticationFacade implements IAuthenticationFacade {
```

Comment:
Why this is Facade but not Service?

Facade usually contains many dependencies on different services. Here are no dependencies at all. Let's make it Service. We already have Authentication Service, why don't you place these methods there?